org.mwc.cmap.legacy/src/MWC/GUI/Layers.java

```diff
@@ -1398,5 +1398,26 @@ public void walkVisibleItems(final Class<?> mustMatch, final OperateFunction fun
 
 		}
 	}
+	
+	public boolean hasNoBounds(Plottable newItem) {
+		boolean hasNoBounds = true;
+		int noBoundsCount=0;
+		Enumeration<Editable> enumer = elements();
+		while(enumer.hasMoreElements()) {
+			Layer item = (Layer)enumer.nextElement();
+			Enumeration<Editable> edits = item.elements();
+			while(edits.hasMoreElements()) {
+				Plottable p = (Plottable)edits.nextElement();
+				if(p.getBounds()!=null) {
+					noBoundsCount++;

```

Comment:
@ayeshare - why do we check that `noBoundsCount` is `> 1`?

Shouldn't we drop out after the first non-null from `p.getBounds()`?

Reply:
If we drop out after the first non-null, then it will never rescale, because, the first time we are adding the very first shape.

Reply:
ahh, I see. Thanks.